File: Serene/Serene.Script/Imports/ServiceContracts/Administration.UserPermissionRow.cs

namespace Serene.Administration
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class UserPermissionRow
    {
        public Int64? UserPermissionId { get; set; }
        public Int32? UserId { get; set; }
        public String PermissionKey { get; set; }
        public String Username { get; set; }
        public String User { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string UserPermissionId = "UserPermissionId";
            [InlineConstant] public const string UserId = "UserId";
            [InlineConstant] public const string PermissionKey = "PermissionKey";
            [InlineConstant] public const string Username = "Username";
            [InlineConstant] public const string User = "User";
        }
    }
    
}

File: Serene/Serene.Web/Modules/Imports/ServiceTypings/Administration.UserPermissionRow.ts
namespace Serene.Administration {
    export interface UserPermissionRow {
        UserPermissionId?: number;
        UserId?: number;
        PermissionKey?: string;
        Grant?: boolean;
        Username?: string;
        User?: string;
    }

    export namespace UserPermissionRow {
        export const idProperty = "UserPermissionId";
        export const nameProperty = "PermissionKey";
        export const localTextPrefix = "Administration.UserPermission";

        export namespace Fields {
            export declare const UserPermissionId: "UserPermissionId";
            export declare const UserId: "UserId";
            export declare const PermissionKey: "PermissionKey";
            export declare const Grant: "Grant";
            export declare const Username: "Username";
            export declare const User: "User";
        }

        ["UserPermissionId","UserId","PermissionKey","Grant","Username","User"].forEach(x => (<any>Fields)[x] = x);
    }
}

